planning/behavior_path_planner/src/util/lane_change/util.cpp

```diff
@@ -762,13 +780,11 @@ PathWithLaneId getReferencePathFromTargetLane(
   const double s_end = std::invoke([&]() {
     const auto dist_from_lc_start = s_start + lane_changing_distance + forward_path_length;
     if (is_goal_in_route) {
-      const auto goal_arc_coordinates =
-        lanelet::utils::getArcCoordinates(target_lanes, route_handler.getGoalPose());
-      const auto dist_to_goal = goal_arc_coordinates.length - min_total_lane_changing_distance;
-      return std::min(dist_from_lc_start, dist_to_goal);
+      const double s_goal =
+        lanelet::utils::getArcCoordinates(target_lanes, route_handler.getGoalPose()).length;
+      return std::min(dist_from_lc_start, s_goal);
     }
-    const auto dist_from_end = target_lane_length - min_total_lane_changing_distance;
-    return std::min(dist_from_lc_start, dist_from_end);
+    return std::min(dist_from_lc_start, target_lane_length);
```

Comment:
As mentioned, this changed created bug for the multiple lane change. Therefore, please help to change the code accordingly.

![Screenshot from 2023-03-24 09-21-39](https://user-images.githubusercontent.com/93502286/227400460-27d32fe8-0131-4d3d-97ac-889656bc77aa.png)

Reply:
Isn't it too short to do the double lane change?